File: java/DeclaraVarOutros.java
/**
 * Programa Java demonstra como declarar variáveis de tipos: boolean, char e String.
 *
 * Objetivos:
 *   - declaração de variáveis;
 *
 * Módulo: Lógica de programação | Capítulo: XXX | Página: XXX
 * Author: Eder Magalhães
 */
public class DeclaraVarOutros {

    public static void main(String[] args) {
        boolean pagamentoAprovado = true;
        
        char letraA = 'a';
        
        String palavra = "Java";
        
        System.out.println("pagamentoAprovado = " + pagamentoAprovado);
        System.out.println("letraA = " + letraA);
        System.out.println("palavra = " + palavra);
        System.out.println();

        int qdteProdutos = 0;

        //atribui a condição boolean
        boolean temEstoque = qdteProdutos > 0;

        //pega o codigo ASCII de 'a' e soma 1
        char letraB = ++letraA;

        //o tipo char suporte nros inteiros (aprox. no limite de short)
        char letraC = 99;

        String frase = "Ola "+palavra;//o operador '+' concatena o conteúdo das strings

        System.out.println("temEstoque " + temEstoque);
        System.out.println("letraB = " + letraB);
        System.out.println("letraC = " + letraC);
        System.out.println("frase = " + frase);
        System.out.println();

        String outraFrase = "Conteudo \'entre aspas\'";
        System.out.println("outraFrase = " + outraFrase);

        //exibe a qtde de letras na frase
        System.out.println("Qtde de letras na outraFrase: " + outraFrase.length());
        System.out.println();

        String texto = "Uma frase no começo.\nOutra no fim...";
        System.out.println("texto = " + texto+"\n");

        String nada = null; //String (objeto) nula
        System.out.println("nada = " + nada);
    }

}

File: python/declaraVarOutros.py
# -*- coding: utf-8 -*-

# Programa Python demonstra como declarar variáveis do tipo String e Boolean.
# Note que em Python não é necessário definir a variável e seu tipo!
# Python é uma linguagem tipada dinamicamente.
#
# Objetivos:
#   - declaração de variáveis que armazenam texto e boolean;
#
# Módulo: Lógica de programação
# Author: Eder Magalhães

pagamentoAprovado = True;

letraA = 'a'; #String com aspas simples

palavra = 'Python';

print "pagamentoAprovado = ", pagamentoAprovado;
print "letraA = ", letraA;
print "palavra = ", palavra;
print;

qdteProdutos = 0;

# atribui a condição boolean
temEstoque = qdteProdutos > 0;

# pega o codigo ASCII de 'a' e soma 1
letraB = chr(ord(letraA)+1);

#converte inteiro para string
letraC = chr(99);

frase = "Ola "+palavra; # o operador '+' concatena conteúdo das strings

print "temEstoque ", temEstoque
print "letraB = ", letraB
print "letraC = ", letraC
print "frase = ", frase
print

outraFrase = "Conteudo 'entre aspas'";
print "outraFrase = ", outraFrase;
print "Qtde de letras na outraFrase: ", len(outraFrase);
print;

texto = """Uma frase no começo.
Outra no fim...""";

print "texto = ", texto, "\n";

nada = None; #String (objeto) nula
print "nada = ", nada;
